### PR TITLE
Feature/split parser

### PR DESCRIPTION
**Pull request by awkay**
I renamed the parameter `:query-interpreter` and added the code to detect and split the parsing. Also removed some dead code and fixed a bug. I'm merging this and pushing it as 2.3.0-SNAPSHOT, and I'll be testing it. Let me know if you see any problems Wilker. Thanks!

### CODE REVIEW

File: src/main/fulcro/client.cljc [hunk 1]
```diff
@@ -184,6 +184,21 @@
...
+(defn- split-parser
+  "Generate a parser that splits reads and writes across two parsers: the supplied query parser for queries, and the built-in
+  parser for mutations."
+  [query-parser]
+  (let [mutation-parser (prim/parser {:read (constantly nil) :mutate app/write-entry-point})]
+    (fn split-parser*
+      ([env query target]
+       (if (mutation-query? query)
+         (mutation-parser env query target)
+         (query-parser (assoc env :target target) query)))
```
Comment (wilkerlucio) [inline src/main/fulcro/client.cljc]: I thought we would only send the custom parser when there is no target, we changed the assumption or was a miss?

Comment (awkay) [reply src/main/fulcro/client.cljc]: this is just to make sure there is a target on there...it will be nil unless I've screwed up

Comment (awkay) [reply src/main/fulcro/client.cljc]: I guess it could just be dropped...I was thinking ppl would expect to see it, but probably not, actually

**Comment by awkay** [comment]
I've removed that little bit of code. Thanks for the review. I tried this out on some basic apps and it seems to be ok, as I would expect.

**Pull request merged**